identity/app/controllers/ResetPasswordController.scala

```diff
@@ -71,11 +71,20 @@ class ResetPasswordController(
   }
 
   def renderResetPassword(token: String, returnUrl: Option[String]): Action[AnyContent] = Action{ implicit request =>
+
+    val isPaymentFailure = (for {
+      url <- returnUrl
+      parsedUrl <- Try(Uri.parse(url)).toOption
+      intcmp <- parsedUrl.query.param("INTCMP")
+    } yield {
+      intcmp.contains("PF") || intcmp.contains("CCX")
```

Comment:
This is probably fine in 99% of cases. Do we care enough about misclassification to use a regex e.g. starts with `PF`. Alternatively, if the only payment failure codes are ` PF, PF1, PF2, PF3, PF4, CCX` (?), then we could create a set of these and just check whether the intcmp is in this set.

Also, worth letting whoever (subscriptions?) know about this dependancy if you haven't already done so.